### ISSUE TITLE
Copy to clipboard image button for all sources 

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#16072]
We have bunch of source code blocks, that people might like to copy and quickly play on local.

We can have overlay image button in right top corner of each source block to copy all content of such source block to clipboard.

Example of long configuration https://checkstyle.org/checks/annotation/annotationlocation.html#Example3-config that would be good to copy in one click.

You can spot how GitHub is doing it for multi line code blocks:
```
1
2
3
```

We need the one more overlay icon to copy config with xml header:
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
```

So users can press such button and get ready to use config. It will help maintainers a lot, when you need to answer questions and you need share config to user that works out of the box

**Comment by yashpandey06** [comment]
Hey @romani,

It seems interesting ...can I take up this issue ? 

Thanks !

**Comment by romani** [comment]
Please send PR.

**Comment by yashpandey06** [comment]
@romani  just confirming ...we are using the htmls from this repo 

https://github.com/checkstyle/checkstyle.github.io/tree/master


and the changes involved for this issue will be in the given repo so ...I need to raise pr there....correct me if i am wrong !

Thanks 1

**Comment by romani** [comment]
all sources are at https://github.com/checkstyle/checkstyle/ , 
to build html pages that you see in github.io use this command - https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only

**Comment by Zopsss** [comment]
@yashpandey06 are you planning to work on this issue? If yes, then please keep in mind the below information.

@romani as I was working on implementing the new skin issue ( https://github.com/checkstyle/checkstyle/pull/16126 ), I saw that the code-blocks' style has also been changed so it's HTML structure. If any contributor provides a solution for the current skin, there's a high chance that it'll not work with the new skin. Maybe should we postpone this issue? 

@yashpandey06 in easier words, we're currently working on upgrading maven-site-plugin which will completely change the look of website, this new site will have new look and feel. It's HTML structure will also change, so if you provide solution which works fine for the current website, there's a high chance that your solution might not work when we upgrade to latest version of maven-site-plugin as it's HTML structure, class & id names will be different than the current site.

**Comment by yashpandey06** [comment]
Not currently

**Comment by yashpandey06** [comment]
Please go  ahead !

**Comment by Zopsss** [comment]
new site is deployed. Contributors can work on this issue now

**Comment by smita1078** [comment]
I am trying out this issue.

> new site is deployed. Contributors can work on this issue now

**Comment by smita1078** [comment]
Ideally what would be preferred, the JavaScript approach that adds button dynamically to all code blocks at runtime or the button to be static (i.e., already present in the HTML when the page is generated)?

**Comment by romani** [comment]
ideally the less JS change html in runtime the better. Lets try with what is easier approach, if it woks ok, we will merge it, and followup with better/reliable/maintainable approach.
same as we did with coloring of code.
I presume that placing static html of button will require update of xdoc generation.

**Comment by smita1078** [comment]
Yes, then I will go first with the easier approach to see how it works!

**Comment by romani** [comment]
@smita1078 , thanks a lot!!!

**Comment by smita1078** [comment]
> [@smita1078](https://github.com/smita1078) , thanks a lot!!!
Most Welcome!
